packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -57,6 +56,154 @@ class NetworkController extends DisposableController
     subscribeToFilterChanges();
   }
 
+  String? exportAsHarFile() {
+    httpRequests =
+        filteredData.value.whereType<DartIOHttpRequestData>().toList();
+
+    if (httpRequests!.isEmpty) {
+      debugPrint('No valid request data to export');
+      return '';
+    }
+
+    try {
+      if (httpRequests!.isNotEmpty) {
+        final har = {
```

Comment:
Is it possible to break this up into multiple flatter representations that are put into the final JSON object later? Building it in a single step makes this a bit difficult to read.

Reply:
split into different parts